package/MDAnalysis/coordinates/PDB.py

```diff
@@ -397,6 +404,7 @@ def _read_frame(self, frame):
 
         pos = 0
         occupancy = np.ones(self.n_atoms)
+        tempfactor = np.zeros(self.n_atoms)
```

Comment:
So a question here - shouldn't the default here be an array of `NaNs` or `None` rather than letting this fail silently if we can't read this column in?

Reply:
Yeah I did wonder that for a bit. I think occupancies defaulting to 1.0 is ok, but maybe tempfactors could be more obviously not present rather than 0?

Reply:
I think for both cases, since folks are using the fields to feed arbitrary data, we probably can't assert that a given default value is ok :/

Reply:
@richardjgowers I should probably say that occupancies is something we should deal with separately, that'll need deprecating etc...

Reply:
I think this is the only blocker to merging this PR. If the record isn't full, this by default will just set unseen records to `0` right? Are we ok with that? It seems like a recipe for disaster given that 0.0 has a meaning with tempfactors (so do occupancies, but I guess we can't fix that without deprecating unless we think it's a bug).

Reply:
En lieu of further views here, please make the default value here `''` as per what we do for incomplete topology attr fields.

i.e. the cases should be:
- Field is seen but not incomplete -> mix of floats & '' for incomplete entries
- Field is fully seen -> floats
- Fied is not seen -> nothing